src/main/scala/temple/builder/project/LanguageConfig.scala

```diff
@@ -0,0 +1,20 @@
+package temple.builder.project
+
+import temple.ast.AttributeType
+
+/** Defines the values used for generation in a language that do not come from the user */
+sealed trait LanguageConfig {
+  val createdByInputName: String
+  val createdByName: String
+  val createdByAttributeType: AttributeType
+}
+
+object LanguageConfig {
+
+  /** The implicit values used to generate Golang code that don't come from the user */
+  object GoLanguageConfig extends LanguageConfig {
+    override val createdByInputName: String            = "authID"
```

Comment:
What is this string actually representing? The database column name? The go struct name?

Reply:
I actually have no idea, @smithwjv explained it to me at one point but I didn't really understand - waiting for an explanation

Reply:
Update: Got a response and have added a comment to the base `LanguageConfig` explaining each param